README
phpMyAdmin - Readme
===================

Version 3.5.5

A set of PHP-scripts to manage MySQL over the web.

http://www.phpmyadmin.net/

Copyright
---------

Copyright (C) 1998-2000
    Tobias Ratschiller <tobias_at_ratschiller.com>

Copyright (C) 2001-2012
    Marc Delisle <marc_at_infomarc.info>
    Olivier Müller <om_at_omnis.ch>
    Robin Johnson <robbat2_at_users.sourceforge.net>
    Alexander M. Turek <me_at_derrabus.de>
    Michal Čihař <michal_at_cihar.com>
    Garvin Hicking <me_at_supergarv.de>
    Michael Keck <mkkeck_at_users.sourceforge.net>
    Sebastian Mendel <cybot_tm_at_users.sourceforge.net>
    [check Documentation.txt/.html file for more details]

License
-------

This program is free software; you can redistribute it and/or modify it under
the terms of the GNU General Public License version 2, as published by the
Free Software Foundation.

This program is distributed in the hope that it will be useful, but WITHOUT
ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
details.

You should have received a copy of the GNU General Public License
along with this program.  If not, see <http://www.gnu.org/licenses/>.

Requirements
------------

* PHP 5.2 or later
* MySQL 5.0 or later
* a web-browser (doh!)

Summary
-------

phpMyAdmin is intended to handle the administration of MySQL over the web.
For a summary of features, please see the Documentation.txt/.html file.

Download
--------

You can get the newest version at http://www.phpmyadmin.net/.

More Information
----------------

Please see the Documentation.txt/.html file.

Support
-------

See reference about support forums under http://www.phpmyadmin.net/


Enjoy!
------

The phpMyAdmin Devel team


PS:

Please, don't send us emails with question like "How do I compile PHP with
MySQL-support". We just don't have the time to be your free help desk.

Please send your questions to the appropriate mailing lists / forums.  Before
contacting us, please read the Documentation.html (esp. the FAQ part).



ChangeLog
phpMyAdmin - ChangeLog
======================

3.5.5.0 (2012-12-21)
- bug #3563824 [export] Support Apache's mod_deflate
- bug #3585523 [interface] Inline query editing broken after row update
- bug #3586389 [setup] Cannot switch language in /setup
- bug #3585695 [CSS] Font size in inline query editor is way too big
- bug #3588354 [l10n] Portuguese Language not displaying correctly
- bug #3591412 [status] Live charts don't work for non-default server
- bug [core] Proxy ajax calls to pma.net to avoid browser notices
- bug #3593534 [tracking] Structure Snapshot on tracked view renders invalid
SQL
- bug #3544366 [events] Event comments not saved

3.5.4.0 (2012-11-16)
- bug #3570212 [edit] uuid_short() is a no-arguments function
- bug #3569577 [edit] Add routine parameter headers not valid for "function"
- bug #3575799 [search] Various search operators not working as expected
- bug #3576322 [search] Invalid select query generated for tables with ENUM fields
- bug #3577468 [display] Incorrect imagejpeg Syntax Breaks Image Transformation
- bug #3578776 [search] Editing SQL not possible when no records found
- bug #3571970 [interface] Display chart and number of rows to plot
- bug #3582631 [core] Wrong redirect url caused cookies error with ForceSSL

3.5.3.0 (2012-10-08)
- bug #3539044 [interface] Browse mode "Show" button gives blank page if no results anymore
- bug #3534979 [interface] Copy Database Ajax feedback vanishes long before copying is done
- bug #3527531 [interface] GC-maxlifetime warning incorrectly displayed
- bug #3526916 [interface] Search fails with JS error when tooltips disabled
- bug #3544366 [interface] Event comments not saved
- bug #3549084 [edit] Can't enter date directly when editing inline
- bug #3548491 [interface] Inline query editor doesn't work from search results
- bug #3547825 [edit] BLOB download no longer works
- bug #3541966 [config] Error in generated configuration arrray
- bug #3553551 [GUI] Invalid HTML code in multi submits confirmation form
- [interface] Designer sometimes places tables on the top menu
- bug #3546277 [core] Call to undefined function __() when config file has wrong permissions
- bug #3540922 [edit] Error searching table with many fields
- bug #3555104 [edit] Cannot copy a DB with table & views
- bug #3559925 [privileges] Incorrect updating of the list of users
- bug #3561224 [edit] cell edit date field with empty date fills in current date
- bug #3559955 [edit] current_date from function drop down fails on update
- bug #3562472 add support for Solaris and FreeBSD system load and memory display in server status
- bug #3553068 [import] Table import from XML file fails
- replace Highcharts with jqplot for Display chart
- bug #3567684 [edit] Pasting value doesn't clear null checkbox
- bug #3570786 [edit] Datepicker for date and datetime fields is broken

3.5.2.2 (2012-08-12)
- [security] Fixed XSS vulnerabilities, see PMASA-2012-4

3.5.2.1 (2012-08-03)
- [security] Fixed local path disclosure vulnerability, see PMASA-2012-3

3.5.2.0 (2012-07-07)
- bug #3521416 [interface] JS error when editing index
- bug #3521313 [core] Call to undefined function __()
- bug #3521016 [edit] NOW() function incorrectly selected
- bug [GUI] Invalid HTML code on transformation_overview.php
- bug #3522930 [browse] Missing validation in Ajax mode
- bug Fix popup message on build SQL of import
- bug #3523499 [core] Make X-WebKit-CSP work better
- replace Highcharts with jqplot for query profiling, zoom search
- bug #3531584 [interface] No form validation in change password dialog
- bug #3531585 [interface] Broken password validation in copy user form
- bug #3531586 [unterface] Add user form prints JSON when user presses enter
- bug #3534121 [config] duplicate line in config.sample.inc.php
- bug #3534311 [interface] Grid editing incorrectly parses ENUM/SET values
- bug #3510196 [core] More clever URL rewriting with ForceSSL

3.5.1.0 (2012-05-03)
- bug #3510784 [edit] Limit clause ignored when sort order is remembered
- bug #3511471 [interface] View name not seen in navi panel (MySQL 5.1)
- bug #3512916 [display] Right frame reloads after displaying SQL result(zero rows)
- bug [interface] Fixed missing Codemirror for inline query edit when exporting a result set
- bug #3514490 [auth] Multiple Navigation panels bug still present
- bug #3515181 [users] Error in create user + underscore + create database
- bug #3515666 [display] Profiling chart shows wrong data
- bug #3516037 [auth] JS includes missing in auth config error page
- bug #3516183 [display] Missing image extension
- bug [display] Added missing icons in original theme
- bug #3516761 [edit] Query error after search
- bug #3516405 [display] Chart title is getting wrong within chart export
- bug #3517021 [interface] Header links except 'More' hide after closing dialog
- bug #3516817 [interface] "More" actions in table structure
- bug #3518484 [privileges] PMA_sqlAddSlashes() does not quote the table names correctly
- bug #3518983 [designer] Error messages do not appear in the Designer
- bug #3519747 [interface] Suhosin patch warning incorrectly displayed
- bug #3520107 [interface] Server status page: Incorrect dialog box titles
- bug #3516089 [structure] DROP does not work on defective VIEWs

3.5.0.0 (2012-04-07)
+ rfe #2021981 [interface] Add support for mass prefix change.
+ "up to date" message on main page when current version is up to date
+ Update to jQuery 1.6.2
+ Patch #3256122 [search] Show/hide db search results
+ Patch #3302354 Add gettext wrappers around a message
+ Remove deprecated function PMA_DBI_get_fields
+ rfe #2098927 Remember recent tables
+ rfe #3078542 Remember the last sort order for each table
+ AJAX for Create table in navigation panel
+ rfe #3310562 Wording about Column
+ AJAX for Add a user in Database privileges

js/tbl_chart.js
var chart_xaxis_idx=-1,chart_series,chart_series_index=-1,temp_chart_title,currentChart=null,chart_data=null,nonJqplotSettings=null,currentSettings=null;
$(document).ready(function(){chart_series="columns";chart_xaxis_idx=$('select[name="chartXAxis"]').attr("value");$("#resizer").resizable({minHeight:240,minWidth:300});$("#resizer").bind("resizestop",function(){$("#querychart").height($("#resizer").height()*0.96);$("#querychart").width($("#resizer").width()*0.96);currentChart.replot({resetAxes:true})});nonJqplotSettings={chart:{type:"line",width:$("#resizer").width()-20,height:$("#resizer").height()-20}};currentSettings={grid:{drawBorder:false,shadow:false,
background:"rgba(0,0,0,0)"},axes:{xaxis:{label:$('input[name="xaxis_label"]').val(),labelRenderer:$.jqplot.CanvasAxisLabelRenderer},yaxis:{label:$('input[name="yaxis_label"]').val(),labelRenderer:$.jqplot.CanvasAxisLabelRenderer}},title:{text:$('input[name="chartTitle"]').attr("value")},legend:{show:true,placement:"outsideGrid",location:"se"}};$("#querychart").html("");$('input[name="chartType"]').click(function(){nonJqplotSettings.chart.type=$(this).attr("value");drawChart();$(this).attr("value")==
"bar"||$(this).attr("value")=="column"?$("span.barStacked").show():$("span.barStacked").hide()});$('input[name="barStacked"]').click(function(){this.checked?$.extend(true,currentSettings,{stackSeries:true}):$.extend(true,currentSettings,{stackSeries:false});drawChart()});$('input[name="chartTitle"]').focus(function(){temp_chart_title=$(this).val()});$('input[name="chartTitle"]').keyup(function(){$(this).attr("value");currentSettings.title=$('input[name="chartTitle"]').val();drawChart()});$('input[name="chartTitle"]').blur(function(){$(this).val()!=
temp_chart_title&&drawChart()});$('select[name="chartXAxis"]').change(function(){chart_xaxis_idx=this.value;drawChart()});$('select[name="chartSeries"]').change(function(){chart_series=this.value;chart_series_index=this.selectedIndex;drawChart()});$('input[name="xaxis_label"]').keyup(function(){currentSettings.axes.xaxis.label=$(this).attr("value");drawChart()});$('input[name="yaxis_label"]').keyup(function(){currentSettings.axes.yaxis.label=$(this).attr("value");drawChart()})});
$("#tblchartform").live("submit",function(){if(!checkFormElementInRange(this,"session_max_rows",PMA_messages.strNotValidRowNumber,1)||!checkFormElementInRange(this,"pos",PMA_messages.strNotValidRowNumber,-1))return false;var a=$(this);if(!checkSqlQuery(a[0]))return false;$(".error").remove();var e=PMA_ajaxShowMessage();PMA_prepareForAjaxRequest(a);$.post(a.attr("action"),a.serialize(),function(b){if(b.success==true){$(".success").fadeOut();if(typeof b.chartData!="undefined"){chart_data=jQuery.parseJSON(b.chartData);
drawChart();$("#querychart").show()}}else{PMA_ajaxRemoveMessage(e);PMA_ajaxShowMessage(b.error,false);chart_data=null;drawChart()}PMA_ajaxRemoveMessage(e)},"json");return false});function drawChart(){nonJqplotSettings.chart.width=$("#resizer").width()-20;nonJqplotSettings.chart.height=$("#resizer").height()-20;currentChart!=null&&currentChart.destroy();currentChart=PMA_queryChart(chart_data,currentSettings,nonJqplotSettings)}
function in_array(a,e){for(var b=0;b<e.length;b++)if(e[b]==a)return true;return false}
function PMA_queryChart(a,e,b){if($("#querychart").length!=0){var d=[],f=[];$.each(a[0],function(h){d.push(h)});switch(b.chart.type){case "column":case "spline":case "line":case "bar":for(var i=0,c=0;c<d.length;c++)if(c!=chart_xaxis_idx){f[i]=[];if(chart_series=="columns"||chart_series==d[c]){$.each(a,function(h,g){f[i].push([g[d[chart_xaxis_idx]],parseFloat(g[d[c]])])});i++}}break;case "pie":if(chart_series!="columns"){f[0]=[];$.each(a,function(h,g){f[0].push([g[d[chart_xaxis_idx]],parseFloat(g[chart_series])])})}}a=
{title:{text:""}};if(b.chart.type=="line")a.axes={xaxis:{},yaxis:{}};if(b.chart.type=="bar"){a.seriesDefaults={renderer:$.jqplot.BarRenderer,rendererOptions:{barDirection:"vertical",highlightMouseOver:true}};a.axes={xaxis:{renderer:$.jqplot.CategoryAxisRenderer},yaxis:{}}}if(b.chart.type=="spline")a.seriesDefaults={rendererOptions:{smooth:true}};if(b.chart.type=="pie")a.seriesDefaults={renderer:$.jqplot.PieRenderer,rendererOptions:{showDataLabels:true,highlightMouseOver:true,showDataLabels:true,dataLabels:"value"}};
$.extend(true,a,e);a.series=[];for(c=0;c<d.length;c++)if(parseInt(chart_xaxis_idx)!=c)if(chart_series=="columns"||chart_series==d[c])a.series.push({label:d[c]});return $.jqplot("querychart",f,a)}};


libraries/language_stats.inc.php
<?php
/* Automatically generated file, do not edit! */
/* Generated by scripts/remove-incomplete-mo */

$GLOBALS["language_stats"] = array (
    'af' => 11,
    'ar' => 51,
    'az' => 17,
    'be@latin' => 35,
    'be' => 35,
    'bg' => 79,
    'bn' => 21,
    'br' => 28,
    'bs' => 19,
    'ca' => 99,
    'cs' => 100,
    'cy' => 27,
    'da' => 100,
    'de' => 100,
    'el' => 100,
    'en_GB' => 100,
    'es' => 100,
    'et' => 99,
    'eu' => 24,
    'fa' => 38,
    'fi' => 74,
    'fr' => 100,
    'gl' => 99,
    'he' => 23,
    'hi' => 56,
    'hr' => 42,
    'hu' => 83,
    'id' => 81,
    'it' => 99,
    'ja' => 99,
    'ka' => 22,
    'ko' => 32,
    'lt' => 72,
    'lv' => 25,
    'mk' => 24,
    'ml' => 2,
    'mn' => 29,
    'ms' => 13,
    'nb' => 76,
    'nl' => 100,
    'pl' => 99,
    'pt_BR' => 97,
    'pt' => 50,
    'ro' => 59,
    'ru' => 100,
    'si' => 80,
    'sk' => 80,
    'sl' => 100,
    'sq' => 30,
    'sr@latin' => 73,
    'sr' => 33,
    'sv' => 100,
    'ta' => 16,
    'te' => 13,
    'th' => 37,
    'tk' => 0,
    'tr' => 100,
    'tt' => 24,
    'ug' => 16,
    'uk' => 55,
    'ur' => 28,
    'uz@latin' => 53,
    'uz' => 53,
    'zh_CN' => 99,
    'zh_TW' => 76,
);
?>


config.inc.php
<?php
/*
 * Generated configuration file
 * Generated by: phpMyAdmin 3.5.5 setup script
 * Date: Fri, 31 May 2013 23:27:53 +0400
 */

/* Servers configuration */
$i = 0;

/* Server: root [2] */
$i++;
$cfg['Servers'][$i]['verbose'] = 'root';
$cfg['Servers'][$i]['host'] = 'localhost';
$cfg['Servers'][$i]['port'] = '';
$cfg['Servers'][$i]['socket'] = '';
$cfg['Servers'][$i]['connect_type'] = 'tcp';
$cfg['Servers'][$i]['extension'] = 'mysqli';
$cfg['Servers'][$i]['nopassword'] = true;
$cfg['Servers'][$i]['auth_type'] = 'config';
$cfg['Servers'][$i]['user'] = 'root';
$cfg['Servers'][$i]['password'] = '';
$cfg['Servers'][$i]['CountTables'] = true;
$cfg['Servers'][$i]['pmadb'] = 'phpmyadmin';
$cfg['Servers'][$i]['controluser'] = 'pma';
$cfg['Servers'][$i]['controlpass'] = 'pmapass';
$cfg['Servers'][$i]['verbose_check'] = false;
$cfg['Servers'][$i]['bookmarktable'] = 'pma_bookmark';
$cfg['Servers'][$i]['relation'] = 'pma_relation';
$cfg['Servers'][$i]['userconfig'] = 'pma_userconfig';
$cfg['Servers'][$i]['table_info'] = 'pma_table_info';
$cfg['Servers'][$i]['column_info'] = 'pma_column_info';
$cfg['Servers'][$i]['history'] = 'pma_history';
$cfg['Servers'][$i]['recent'] = 'pma_recent';
$cfg['Servers'][$i]['table_uiprefs'] = 'pma_table_uiprefs';
$cfg['Servers'][$i]['tracking'] = 'pma_tracking';
$cfg['Servers'][$i]['table_coords'] = 'pma_table_coords';
$cfg['Servers'][$i]['pdf_pages'] = 'pma_pdf_pages';
$cfg['Servers'][$i]['designer_coords'] = 'pma_designer_coords';
$cfg['Servers'][$i]['AllowNoPassword'] = true;

/* Server: host [3] */
$i++;
$cfg['Servers'][$i]['verbose'] = 'host';
$cfg['Servers'][$i]['host'] = 'localhost';
$cfg['Servers'][$i]['port'] = '';
$cfg['Servers'][$i]['socket'] = '';
$cfg['Servers'][$i]['connect_type'] = 'tcp';
$cfg['Servers'][$i]['extension'] = 'mysqli';
$cfg['Servers'][$i]['auth_type'] = 'cookie';
$cfg['Servers'][$i]['user'] = '';
$cfg['Servers'][$i]['password'] = '';
$cfg['Servers'][$i]['AllowRoot'] = false;
$cfg['Servers'][$i]['CountTables'] = true;

/* End of servers configuration */

$cfg['UploadDir'] = '';
$cfg['SaveDir'] = '';
$cfg['BZipDump'] = false;
$cfg['DefaultLang'] = 'ru';
$cfg['ThemeDefault'] = 'original';
$cfg['ServerDefault'] = 1;
$cfg['CompressOnFly'] = false;
$cfg['UserprefsDeveloperTab'] = true;
$cfg['HideStructureActions'] = false;
$cfg['LoginCookieDeleteAll'] = false;
$cfg['QueryHistoryDB'] = true;
$cfg['RetainQueryBox'] = true;
$cfg['blowfish_secret'] = '51a360783193d3.45092927';
$cfg['LeftDefaultTabTable'] = 'tbl_select.php';
$cfg['MaxTableList'] = 500;
?>